TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -771,6 +771,32 @@ public class DataDescription: DataDefinition, Parent<DataDescription>{
         [NotNull]
         public new DataDescriptionEntry CodeElement => (DataDescriptionEntry) base.CodeElement;
 
+        public bool IsDataDescriptionGroup
+        {
+            get
+            {
+                if (this.CodeElement?.LevelNumber?.Value < 50 &&
+                    this.TypeDefinition == null && this.Picture == null && !this.CodeElement.ConsumedTokens.Any(ct => ct.SourceText.Equals("Type", StringComparison.OrdinalIgnoreCase)))
```

Comment:
TypeDefinition can be null if the type cannot be resolved.
Consumed tokens  are very low level information.

The best way to test if a a CodeElement contains the "type" keyword is to test UserDefinedDataType property of class CommonDataDescriptionAndDataRedefines.

Reply:
It works on types that hasn't been defined, but not on cases like "01 MyToto Type PIC X.". The value of UserDefinedDataType is null

Reply:
Done, see eddc61e2108221d74c72393bf911fa0a57408ce2.